packages/entities/entities-plugins/src/components/PluginForm.vue

```diff
@@ -0,0 +1,236 @@
+import type {
+  KonnectPluginFormConfig,
+  KongManagerPluginFormConfig,
+  PluginFormState,
+  PluginFormFields,
+} from '../types'
+import endpoints from '../plugins-endpoints'
+// import composables from '../composables'
+import { useAxios, useErrors, /* EntityFormSection, */ EntityBaseForm, EntityBaseFormType } from '@kong-ui-public/entities-shared'
+import '@kong-ui-public/entities-shared/dist/style.css'
+
+const emit = defineEmits<{
+  (e: 'update', data: Record<string, any>): void,
+  (e: 'error', error: AxiosError): void,
+  (e: 'loading', isLoading: boolean): void,
+}>()
+
+// Component props - This structure must exist in ALL entity components, with the exclusion of unneeded action props (e.g. if you don't need `canDelete`, just exclude it)
+const props = defineProps({
+  /** The base konnect or kongManger config. Pass additional config props in the shared entity component as needed. */
+  config: {
+    type: Object as PropType<KonnectPluginFormConfig | KongManagerPluginFormConfig>,
+    required: true,
+    validator: (config: KonnectPluginFormConfig | KongManagerPluginFormConfig): boolean => {
+      if (!config || !['konnect', 'kongManager'].includes(config?.app)) return false
+      if (config.app === 'konnect' && !config.controlPlaneId) return false
+      if (config.app === 'kongManager' && typeof config.workspace !== 'string') return false
+      if (!config.cancelRoute) return false
+      return true
+    },
+  },
+  /** If a valid Plugin ID is provided, it will put the form in Edit mode instead of Create */
+  pluginId: {
+    type: String,
+    required: false,
+    default: '',
+  },
+})
+
+const router = useRouter()
+// const { i18n: { t } } = composables.useI18n()
+const { getMessageFromError } = useErrors()
+
+const { axiosInstance } = useAxios({
+  headers: props.config?.requestHeaders,
+})
+
+const fetchUrl = computed<string>(() => endpoints.form[props.config.app].edit)
+const formType = computed((): EntityBaseFormType => props.pluginId ? EntityBaseFormType.Edit : EntityBaseFormType.Create)
+
+const form = reactive<PluginFormState>({
+  fields: {
+    name: '',
+    tags: '',
+    entity_id: '',
+  },
+  isReadonly: false,
+  errorMessage: '',
+})
+
+const formFieldsOriginal = reactive<PluginFormFields>({
+  name: '',
+  tags: '',
+  entity_id: '',
+})
+
+/**
+ * Is the form submit button enabled?
+ * If the form.fields and formFieldsOriginal are equal, always return false
+ */
+const canSubmit = computed((): boolean => JSON.stringify(form.fields) !== JSON.stringify(formFieldsOriginal) && !!form.fields.name)
+
+const initForm = (data: Record<string, any>): void => {
+  form.fields.name = data?.name || ''
+  form.fields.tags = data?.tags?.join(', ') || ''
+  // TODO:
+  // form.fields.entity_id = data?.certificate?.id || ''
+
+  // Set initial state of `formFieldsOriginal` to these values in order to detect changes
+  Object.assign(formFieldsOriginal, form.fields)
```

Comment:
You may want to check that things remain reactive and are correctly mapped here -- probably fine, but double-check